tau +rocm w/ ROCm 6.1.1: Could not find a package configuration file provided by "LLVM" with any of the following names... LLVMConfig.cmake, llvm-config.cmake

### Steps to reproduce the issue

`tau@2.33.1 +rocm` build fails using:
* spack@develop (641075539c0a888d4378bd7dfdb19a872c5c42de from `Mon Jun 3 14:37:51 2024 -0700`)
* ROCm 6.1.1 installed via apt, externally defined via `packages:` config
* Ubuntu 22.04, X86_64

Spack environment: [spack.yaml.txt](https://github.com/user-attachments/files/15778562/spack.yaml.txt)

<details><summary>Concretization</summary>

<pre>
 -   fkxhq4p  tau@2.33.2%gcc@11.4.0~adios2+binutils~comm~craycnl~cuda~disable-no-pie~dyninst+elf+fortran~gasnet+io~level_zero+libdwarf+libunwind~likwid+mpi~ompt~opari~opencl~openmp+otf2+papi+pdt~phase~ppc64le~profileparam+pthreads~python+rocm+rocprofiler~rocprofv2~roctracer~scorep~shmem~sqlite+syscall~x86_64 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  ayjqbjz      ^binutils@2.42%gcc@11.4.0~gas~gprofng+headers~interwork~ld+libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  sycoqix          ^diffutils@3.10%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  mv2wke5          ^pkgconf@2.2.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  fz3fbgw          ^zstd@1.5.6%gcc@11.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  m2fbj54      ^elfutils@0.191%gcc@11.4.0~debuginfod+exeprefix+nls build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  tmq3cku          ^bzip2@1.0.8%gcc@11.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  3ea5rwn          ^gettext@0.22.5%gcc@11.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  a4yvhcg              ^tar@1.34%gcc@11.4.0 build_system=autotools zip=pigz arch=linux-ubuntu22.04-x86_64_v3
[+]  5qg7ajj                  ^pigz@2.8%gcc@11.4.0 build_system=makefile arch=linux-ubuntu22.04-x86_64_v3
[+]  iloyaza          ^m4@1.4.19%gcc@11.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-x86_64_v3
[+]  jv4abt3              ^libsigsegv@2.14%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  imwcbq7          ^xz@5.4.6%gcc@11.4.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  hshzy76      ^gcc-runtime@11.4.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[e]  a7drdl4      ^glibc@2.35%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  aevr47s      ^gmake@4.4.1%gcc@11.4.0~guile build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[e]  qyqwwse      ^hsa-rocr-dev@6.1.1%gcc@11.4.0~asan+image+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[+]  ok74pjo      ^hwloc@2.9.3%gcc@11.4.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  cinntmr          ^libpciaccess@0.17%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  ekglkxs              ^libtool@2.4.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  eu34nvi              ^util-macros@1.20.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  blvm6hb          ^libxml2@2.10.3%gcc@11.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  sqasgz3          ^ncurses@6.5%gcc@11.4.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu22.04-x86_64_v3
[+]  22frrec      ^libdwarf@20180129%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  nlwa37j      ^libunwind@1.6.2%gcc@11.4.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  hyhvy5g      ^mpich@4.2.1%gcc@11.4.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath~xpmem build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=default arch=linux-ubuntu22.04-x86_64_v3
[+]  43jtbdg          ^findutils@4.9.0%gcc@11.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu22.04-x86_64_v3
[+]  grokknt          ^libfabric@1.21.0%gcc@11.4.0~debug~kdreg~uring build_system=autotools fabrics=sockets,tcp,udp arch=linux-ubuntu22.04-x86_64_v3
[+]  rc3e5qf          ^yaksa@0.3%gcc@11.4.0~cuda~rocm build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  tfy5aww              ^autoconf@2.72%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  vqvzn64                  ^perl@5.38.2%gcc@11.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-x86_64_v3
[+]  sw7gp6x                      ^berkeley-db@18.1.40%gcc@11.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-x86_64_v3
[+]  c5h3rwv              ^automake@1.16.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  gl6u6pc              ^python@3.11.9%gcc@11.4.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu22.04-x86_64_v3
[+]  6sx3gbp                  ^expat@2.6.2%gcc@11.4.0+libbsd build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  wfvladn                      ^libbsd@0.12.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  i3snxv7                          ^libmd@1.0.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  xucr42b                  ^gdbm@1.23%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  ounz353                  ^libffi@3.4.6%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  wq6feyo                  ^libxcrypt@4.4.35%gcc@11.4.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu22.04-x86_64_v3
[+]  r3wk5bb                  ^openssl@3.3.0%gcc@11.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-x86_64_v3
[+]  t2brqss                      ^ca-certificates-mozilla@2023-05-30%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  bo6z4bc                  ^readline@8.2%gcc@11.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-x86_64_v3
[+]  jvljkqx                  ^sqlite@3.43.2%gcc@11.4.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  vl2hqwb                  ^util-linux-uuid@2.40.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  yzhiwoi      ^openjdk@11.0.23_9%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  mogwqv7      ^otf2@3.0.3%gcc@11.4.0 build_system=autotools patches=7e56d93 arch=linux-ubuntu22.04-x86_64_v3
[+]  fgtywmi      ^papi@7.1.0%gcc@11.4.0~cuda~debug+example~infiniband~lmsensors~nvml~powercap~rapl~rocm~rocm_smi~sde+shared~static_tools build_system=autotools patches=48cb202 arch=linux-ubuntu22.04-x86_64_v3
[+]  sjb6txo      ^pdt@3.25.2%gcc@11.4.0~pic build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[e]  bypjxzz      ^rocm-smi-lib@6.1.1%gcc@11.4.0~asan+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  ziuv62l      ^rocprofiler-dev@6.1.1%gcc@11.4.0 build_system=cmake build_type=Release generator=make patches=9f49746 arch=linux-ubuntu22.04-x86_64_v3
[+]  54554gr      ^zlib-ng@2.1.6%gcc@11.4.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
</pre>
</details>

Error:
```
$> spack -e . install
...
==> Installing tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl [50/50]
==> No binary for tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl found: installing from source
==> Fetching https://www.cs.uoregon.edu/research/tau/tau_releases/tau-2.33.2.tar.gz
==> No patches needed for tau
==> tau: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'install'

15 errors found in build log:
     94     TAU: PDT: Using binary rewriting capabilities from PEBIL in tau_pebil_rewrite
     95     ====================================================
     96      Copy /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/tools/src/contrib/CubeReader.jar to /spack/opt/spack/linux-ubunt
            u22.04-x86_64_v3/gcc-11.4.0/tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/x86_64/lib
     97     ====================================================
     98     ... done
     99     ***********************************************************************
  >> 100    grep: ./utils/FixMakefile.info: No such file or directory
  >> 101    grep: ./utils/FixMakefile.sed: No such file or directory
     102    NOTE: Using ELF support
     103    NOTE: Using the TAU's ROCm Options
     104    NOTE: Using the TAU's ROCm ROCPROFILER Interface
     105    NOTE: Using TAU's ROCm v6
     106    NOTE: Saving configuration environment to /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/.configure_env/321da39cd1b18
            44ba1abbd64801420e2
     107    NOTE: Enabled Profiling. Compiling with -DPROFILING_ON

     ...

     238    rm -f -f scalls.o tracee.o ptrace_syscall.o libTAU-ptrace-syscall.so
     239    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/src/wrappers/syscall'
     240    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/instrument'
     241    /bin/rm -f simple.o simple
     242    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/instrument'
     243    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/threads'
  >> 244    Makefile:16: /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocprofiler-rocmv6-papi-mpi-
            pthread-pdt: No such file or directory
  >> 245    make[1]: *** No rule to make target '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocp
            rofiler-rocmv6-papi-mpi-pthread-pdt'.  Stop.
     246    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/threads'
     247    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/cthreads'
  >> 248    Makefile:16: /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocprofiler-rocmv6-papi-mpi-
            pthread-pdt: No such file or directory
  >> 249    make[1]: *** No rule to make target '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocp
            rofiler-rocmv6-papi-mpi-pthread-pdt'.  Stop.
     250    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/cthreads'
     251    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/fortran'
     252    /bin/rm -f hello.o hello
     253    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/fortran'
     254    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/f90'
     255    /bin/rm -f cubes.o cubes
     256    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/f90'
     257    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/pi'
  >> 258    Makefile:16: /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocprofiler-rocmv6-papi-mpi-
            pthread-pdt: No such file or directory
  >> 259    make[1]: *** No rule to make target '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocp
            rofiler-rocmv6-papi-mpi-pthread-pdt'.  Stop.
     260    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/pi'
     261    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/NPB2.3'
     262    rm -f core
     263    rm -f *~ */core */*~ */*.o */npbparams.h */*.obj */*.exe
     264    rm -f MPI_dummy/test MPI_dummy/libmpi.a
     265    rm -f sys/setparams sys/makesuite sys/setparams.h
     266    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/NPB2.3'
     267    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/autoinstrument'
  >> 268    Makefile:16: /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocprofiler-rocmv6-papi-mpi-
            pthread-pdt: No such file or directory
  >> 269    make[1]: *** No rule to make target '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocp
            rofiler-rocmv6-papi-mpi-pthread-pdt'.  Stop.
     270    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/autoinstrument'
     271    make[1]: Entering directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/reduce'
  >> 272    Makefile:16: /tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocprofiler-rocmv6-papi-mpi-
            pthread-pdt: No such file or directory
  >> 273    make[1]: *** No rule to make target '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/x86_64/lib/Makefile.tau-rocm-rocp
            rofiler-rocmv6-papi-mpi-pthread-pdt'.  Stop.
     274    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/examples/reduce'
     275    touch .clean
     276    Determining Configuration...
     277    System previously configured as a x86_64
     278    *********** RECURSIVELY MAKING SUBDIRECTORIES ***********
     279    *** COMPILING src/TraceInput DIRECTORY

     ...

     881    -- Detecting C compile features - done
     882    -- Detecting CXX compiler ABI info
     883    -- Detecting CXX compiler ABI info - done
     884    -- Check for working CXX compiler: /usr/bin/g++ - skipped
     885    -- Detecting CXX compile features
     886    -- Detecting CXX compile features - done
  >> 887    CMake Error at CMakeLists.txt:14 (find_package):
     888      Could not find a package configuration file provided by "LLVM" with any of
     889      the following names:
     890
     891        LLVMConfig.cmake
     892        llvm-config.cmake
     893

     ...

     895      "LLVM_DIR" to a directory containing one of the above files.  If "LLVM"
     896      provides a separate development package or SDK, be sure it has been
     897      installed.
     898
     899
     900    -- Configuring incomplete, errors occurred!
  >> 901    make[1]: *** [Makefile:11: install] Error 1
     902    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-2.33.2-fkxhq4p5jlhm7moecuc2qngnlcydqrpl/spack-src/plugins/llvm'
  >> 903    make: *** [Makefile:195: install] Error 2
```

### Error message

<details><summary>Error message</summary>

<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (136308c8fdc3d3d0593d563103f34e60ab15a5d6)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/15778565/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/15778566/spack-build-out.txt)

@khuck @sameershende @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/tau/package.py
+++ b/var/spack/repos/builtin/packages/tau/package.py
@@ -178,6 +178,9 @@ class Tau(Package):
         msg="Using ROCm, select either +rocprofiler, +roctracer or +rocprofv2",
     )
 
+    # https://github.com/UO-OACISS/tau2/commit/1d2cb6b
+    patch("tau-rocm-disable-llvm-plugin.patch", when="@2.33.2 +rocm")
+
     filter_compiler_wrappers("Makefile", relative_root="include")
     filter_compiler_wrappers("Makefile.tau*", relative_root="lib")
     filter_compiler_wrappers("Makefile.tau*", relative_root="lib64")

--- a/var/spack/repos/builtin/packages/tau/tau-rocm-disable-llvm-plugin.patch
+++ b/var/spack/repos/builtin/packages/tau/tau-rocm-disable-llvm-plugin.patch
@@ -1,0 +1,19 @@
+diff -ruN spack-src/configure spack-src-patched/configure
+--- spack-src/configure	2024-04-16 00:20:41.000000000 +0000
++++ spack-src-patched/configure	2024-06-12 19:35:50.799407280 +0000
+@@ -3101,7 +3101,6 @@
+             rocprofiler=yes
+             tauoptions="${tauoptions}-rocm"
+             pthread=yes
+-            build_llvm_plugin=yes
+             shift
+             ;;
+ 
+@@ -3118,7 +3117,6 @@
+             rocprofiler=yes
+             tauoptions="${tauoptions}-rocm"
+             pthread=yes
+-            build_llvm_plugin=yes
+             shift
+             ;;
+ 

